### PR TITLE
[DRFT-371] Dispatch export notifications 

### PR DESCRIPTION
**Pull request by johnsonm325**
This code change adds proper notifications and corresponding language to exporting of tables. Check the DRFT-371 card to see the documentation on which notification should appear on exporting. We should have an "in progress", "success" and "error".

To test, export any of the tables in drift. To test errors, mess up the code that handles exporting and check the error. You can find that code here: https://github.com/RedHatInsights/drift-frontend/blob/master/src/SmartComponents/helpers.js#L86 and here: https://github.com/RedHatInsights/drift-frontend/blob/master/src/SmartComponents/modules/helpers.js#L509

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by Fewwy** [comment]
@johnsonm325 I downloaded your branch and I don't think it's working as intended.
1. When I click to export the table in /beta/insights/drift/baselines I download the file 3 times
2. There is no notification about "success", only "downloading export"

**Comment by johnsonm325** [comment]
@Fewwy If you check the document in the Jira card that I referenced, you'll see what the language should be. It doesn't say success, it says "Downloading export". So it is correct. https://docs.google.com/document/d/1_5Gkfz9SaggEk7l5Cln4oGitHyRd8D1LO5Yu9KM40Ag/edit#heading=h.c8dlkqyo96mk

If you'll notice though, downloading the export on the baselines table does create 3 files, which is not right. So I need to fix that.

**Comment by Fewwy** [comment]
@johnsonm325 Aha, that's correct, sorry I got as far as finding the type of notification and not the message itself :DDDD
Ping me here when you fix the "download 3 files" issue, I will test it :)

**Comment by johnsonm325** [comment]
@Fewwy This is ready for another look

**Comment by Fewwy** [comment]
@johnsonm325 Sorry for the late re-review. Somehow I missed the notification :(
I'm downloading CSV, it works good
I have a question - what is this last column in the CSV? It has no name just "true,true,true"
image

**Comment by Fewwy** [comment]
I don't know if the notification about success should fade after some time. Mine persisted and won't disappear
image

**Comment by Fewwy** [comment]
Found one more difference, If I'm on the single baseline page and Downloading CSV - it has the baseline name + 2 columns 
image
If I download JSON for the same baseline the name of the baseline is missing from the JSON object structure. 
image

**Comment by johnsonm325** [comment]
> Found one more difference, If I'm on the single baseline page and Downloading CSV - it has the baseline name + 2 columns image If I download JSON for the same baseline the name of the baseline is missing from the JSON object structure. image

This is the expected behavior. It's how we designed it from the beginning. Good eye though.

**Comment by johnsonm325** [comment]
@Fewwy this is ready for a re-review. I made 2 of the 3 changes you suggested.

File: src/SmartComponents/BaselinesTable/redux/__tests__/baselinesTableReducer.fixtures.js [hunk 1]
```diff
@@ -92,15 +92,15 @@ function baselineTableDataOneSelected() {
     return baselineTableDataOneSelected;
 }
 
-const tableDataCSV = 'UUID,Name,Last updated,Associated systems\n\
-1234,beavs baseline,4 years ago,3,true\n\
-abcd,micjohns baseline,4 years ago,0,false\n\
+const tableDataCSV = 'UUID,Name,Last updated,Associated systems,Notifications enabled\n\
+1234,beavs baseline,3 years ago,3,true\n\
```
Comment (bastilian) [inline src/SmartComponents/BaselinesTable/redux/__tests__/baselinesTableReducer.fixtures.js:96]: We should do something about these tests here. This is now the third PR or so where it pops up and is unrelated to the PR itself. We need to look into mocking time here and make sure that it'll always be 3 or 4 years, or both, but it should never change over time.

Comment (johnsonm325) [reply src/SmartComponents/BaselinesTable/redux/__tests__/baselinesTableReducer.fixtures.js:96]: those are now fixed here: https://github.com/RedHatInsights/drift-frontend/pull/616

**Comment by johnsonm325** [comment]
@Fewwy @bastilian I'll take another review here.

**Comment by Fewwy** [comment]
@johnsonm325 Reviewed it again, works as intented, LGTM

**Review by Fewwy — APPROVED** [review]

**Pull request merged**